detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryAbstractClassSpec.kt

```diff
@@ -88,6 +88,21 @@ class UnnecessaryAbstractClassSpec(val env: KotlinCoreEnvironment) {
                 """
                 assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
             }
+
+            @Test
+            fun `does not report abstract class that inherits from an interface and an abstract class in that order`() {
+                val code = """
+                interface I
+
+                @Deprecated("We don't care about this first class")
+                abstract class A {
+                    abstract val i: Int
+                }
+                abstract class B: A(), I
```

Comment:
```suggestion
                abstract class B: A(), I
                abstract class C: I, A()
```

We should check both cases.

Reply:
Sure! Let me add another test case as well!

Reply:
@gouri-panda I have an idea to solve this in a generic manner. If you're okay I can raise a PR on your fork branch `gouri-panda:fix-4753`?

Reply:
I was thinking if we check one of the parent is abstract, then we can return. But go for it :)

Reply:
@gouri-panda my idea was to check if all parents are interfaces, basically the complementary set of whether any parent is abstract. The reason being I'm unable to find any helper method indicating if the parent class is abstract.

Reply:
@gouri-panda 
I have pushed my changes as a [PR to your fork's branch here](https://github.com/gouri-panda/detekt/pull/1).
In summary I have checked if not all the parents are interfaces, which is equivalent of the fact that at least one parent is abstract.
The [commit is here](https://github.com/gouri-panda/detekt/pull/1/commits/69b3fd32157890e824e84781471fc660a14a19fd) (there are a few other merge commits)